Graph DAO with tests

mattsooknah opened the pull request:
First stab at a graph DAO using Tinkerpop and OrientDB. Also cleaned up the dataaccess package. 

Does not include: using remote DB server, smart transactions.

dvoet commented on src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala:
```diff
@@ -0,0 +1,106 @@
+import scala.collection.JavaConverters._
+
+class GraphEntityDAO(backingDatabase: Graph) extends EntityDAO with GraphDAO {
+  val db = backingDatabase
```
why do we have this val if it is the same as the constructor param?

dvoet commented on src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala:
```diff
@@ -0,0 +1,50 @@
+
+  def getEntityVertex(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String): Option[Vertex] = {
+    def entityFunc = hasProperty("_name", entityName)
+    getWorkspaceVertex(workspaceNamespace, workspaceName).flatMap(w => {
```
unless my understanding is wrong, this is doing 2 db hits, one to get the workspace then one to get the entity. Can we do this in 1?

dvoet commented on src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala:
```diff
@@ -0,0 +1,50 @@
+
+  def isReservedProperty(key: String) = key.startsWith("_")
+
+  def getWorkspaceVertex(namespace: String, name: String): Option[Vertex] = {
```
it does not seem right to have getWorkspaceVertex and getEntityVertex here. Perhaps instead we have a catalog of filter functions... essentially named queries

dvoet commented on src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala:
```diff
@@ -0,0 +1,106 @@
+
+  def save(workspaceNamespace: String, workspaceName: String, entity: Entity): Entity = {
+    val entityName = entity.name
+    val entityType = entity.entityType
```
why have these vals?

mattsooknah commented:
I've addressed the comments. It's a bit more elegant now (e.g. passing pipelines around instead of vertices, so we only hit the DB once). Also I added proper attribute serialization (i.e. store properties as their actual Java types instead of Object).
